src/PostgREST/ApiRequest.hs

```diff
@@ -125,8 +126,11 @@ userApiRequest schema req reqBody
     case decodeContentType . fromMaybe "application/json" $ lookupHeader "content-type" of
       CTApplicationJSON ->
           either Left (\val -> case ensureUniform (pluralize val) of
-            Nothing -> Left "All object keys must match"
-            Just json -> Right json) (JSON.eitherDecode reqBody)
+              Nothing -> Left "All object keys must match"
+              Just json -> Right json)
+            (if BL.null reqBody && isTargetingProc
+              then Right emptyObject
+              else JSON.eitherDecode reqBody)
```

Comment:
Your change got me thinking...what if we use the monad instance of Either to clean this code up?

```haskell
CTApplicationJSON -> do
  val <- if BL.null reqBody && isTargetingProc
           then Right emptyObject
           else JSON.eitherDecode reqBody
  case ensureUniform (pluralize val) of
    Nothing -> Left "All object keys must match"
    Just json -> Right json
```

What's your opinion? Do you like it better written that way? If so you could conver the `CTTextCSV` case below to this style as well.